src/aws-elasticache/user-group.ts

```diff
@@ -0,0 +1,166 @@
+import { IResource, Lazy, Names, Resource, Stack, Token, aws_elasticache } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { IUser } from './user';
+import { Engine } from './util';
+
+/**
+ * An User Group
+ */
+export interface IUserGroup extends IResource {
+  /**
+   * The ARN of the user group.
+   *
+   * @attribute
+   */
+  readonly userGroupArn: string;
+  /**
+   * The ID of the user group.
+   *
+   * @attribute
+   */
+  readonly userGroupId: string;
+}
+
+/**
+ * Properties for defining an User Group.
+ */
+export interface UserGroupProps {
+  /**
+   * The ID of the user group.
+   *
+   * \`userGroupId\` can have up to 40 characters.
+   *
+   * \`userGroupId\` must consist only of alphanumeric characters or hyphens, with the first character as a letter, and it can't end with a hyphen or contain two consecutive hyphens.
```

Comment:
This constraint is quoted from the management console. It is not written in the [CFn documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-usergroup.html#cfn-elasticache-usergroup-usergroupid).

<img width="764" alt="image" src="https://github.com/user-attachments/assets/cdba97ed-9591-45de-bb00-94bbed7bfff6">